handlers.go

```diff
@@ -47,18 +57,20 @@ func hashSHA256AsHex(toHash []byte) string {
 	return fmt.Sprintf("%X", h.Sum(nil))
 }
 
-func logSigningRequestFailure(sigreq formats.SignatureRequest, sigresp formats.SignatureResponse, rid, userid, inputHash string, starttime time.Time, err error) {
+func logSigningRequestFailure(sigreq formats.SignatureRequest, sigresp formats.SignatureResponse, rid, userid, inputHash string, inputHashes []string, starttime time.Time, err error) {
 	log.WithFields(log.Fields{
-		"rid":         rid,
-		"options":     sigreq.Options,
-		"mode":        sigresp.Mode,
-		"ref":         sigresp.Ref,
-		"type":        sigresp.Type,
-		"signer_id":   sigresp.SignerID,
-		"input_hash":  inputHash,
-		"output_hash": nil,
-		"user_id":     userid,
-		"t":           int32(time.Since(starttime) / time.Millisecond), //  request processing time in ms
+		"rid":           rid,
```

Comment:
nit: Seeing this set of fields in a number of locations, I wonder if this should be moved to a struct that can output a `log.Fields`.

Reply:
Yeah, it's not exactly DRY. The [logrus readme](https://github.com/sirupsen/logrus#example) includes this example:

```go
  // A common pattern is to re-use fields between logging statements by re-using
  // the logrus.Entry returned from WithFields()
  contextLogger := log.WithFields(log.Fields{
    "common": "this is a common field",
    "other": "I also should be logged always",
  })
```

so that might be an option too. Unless you think it should be addressed in this PR I'll take a TODO and file an issue to clean that up.